pkg/services/ngalert/migration/alert_rule.go

```diff
@@ -304,3 +303,12 @@ func extractChannelIDs(d *migrationStore.DashAlert) (channelUids []migrationStor
 
 	return channelUids
 }
+
+// simpleTitleDedup truncates the given title to the maximum length and appends a simple suffix based on dashboard and panel id to make it unique.
+// This could fail to be unique if identically named alerts in different dashboards are migrating to the same folder. However, in most cases
+// this is sufficient and more readable appending a random uid.
+func simpleTitleDedup(da *migrationStore.DashAlert, title string) string {
+	suffix := fmt.Sprintf("_%d-%d", da.DashboardID, da.PanelID)
```

Comment:
Id seems to be ok-ish here. However, it is not used anywhere in the UI and even API. IIRC, we address the dashboard by UID everywhere. I wonder if it could be even more simpler to use just simple sequential index as a suffix?

Reply:
Great idea, done here: [aa745c9](https://github.com/grafana/grafana/pull/78351/commits/aa745c9fb06c2382d59ccc093e3c4499858ba352)

Wanted to add better testing here so it made sense to encapsulate more of the logic into the deduplicator.

Note that channel UID migration was also using it, but the reason it did is no longer relevant. In the old migration we had to consider that a channel might not have a UID, now we are guaranteed a unique UID for all channels so I chose to remove that part instead of making `Deduplicator.contains` public.